daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/CharsetUtils.scala

```diff
@@ -33,11 +33,13 @@ object CharsetUtils {
    * encodings as well as the standard ones.
    */
   def getCharset(name: String): BitsCharset = {
-    val cs = DaffodilCharsetProvider.charsetForName(name)
-    cs
+    val cs = BitsCharsetDefinitionRegistry.find(name).getOrElse(null)
+    if (cs == null)
+      SDE("The encoding '%s' was not found. Available choices are: %s", name, BitsCharsetDefinitionRegistry.supportedEncodingsString)
```

Comment:
This changes the behavior of this function. In the past it returned null if the encoding wasn't found and the caller was expected to handle it correctly. I'd guess we want to keep that behavior, since some callers might just want to know if a charset exists or not, and do something other than error if not. 

Maybe something like this instead?
```scala
val cs = BitsCharsetDefinitionRegistry.find(name).map(_.newInstance).getOrElse(null)
cs
```

Reply:
I think this SDE should be removed. This function should either return a charset or null. The callers will determine how to handle null.